authelia: init at 4.37.5

06kellyjac opened the pull request:
###### Description of changes

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

Package authelia for nixpkgs

go built from src + ~prebuilt html bundle~ html bundle now built from source
updater script to keep the two updated in tandem

related #180381 #165345 #207931

cc: @LongerHV
I'd be more than happy to maintain this with you. I can't find you in the existing maintainer list, or added in that PR.
After this merges feel free to add yourself to the main maintainers list in a comment & to this package

**Update:**
Full source build, service added, nixos test added
thanks all who assisted

TODO: add some real documentation lines to the service
~probably get rid of the 2nd "client" vm and just do the requests on the authelia vm~ done

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [X] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [X] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [X] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

AndersonTorres commented on pkgs/servers/authelia/default.nix:
```diff
@@ -0,0 +1,81 @@
+
+  subPackages = [ "cmd/authelia" ];
+
+  ldflags = let p = "github.com/authelia/authelia/v${lib.versions.major version}/internal/utils"; in [
```
```suggestion
  ldflags = let 
    p = "github.com/authelia/authelia/v${lib.versions.major version}/internal/utils"; 
  in [
```

RaitoBezarius replied:
I think the suggestion was not applied?

06kellyjac replied:
done now

06kellyjac commented on pkgs/servers/authelia/default.nix:
```diff
@@ -0,0 +1,81 @@
+    rev = "v${version}";
+    sha256 = "sha256-W0xkm/rsFLg34Amx4LuR21+2EsLVSGRa5D7VoM3THgw=";
+  };
+  vendorSha256 = "sha256-6EVEP7eIwW8rQIoV3N4qjr8HsS6+yTlD3Ox19Y8GL4w=";
```
No thank you

pinpox commented:
@AndersonTorres The requested changes seem to have been applied. What is the state on this? I'd like to use authelia on nixos

06kellyjac commented:
I should probably rebase it by now since it was probably built before the move to go1.18

pinpox commented:
> I should probably rebase it by now since it was probably built before the move to go1.18

That would be great, let me know if you need help.

AndersonTorres commented:
> @AndersonTorres The requested changes seem to have been applied. What is the state on this? I'd like to use authelia on nixos

As 06kellyjac is working on it, I will not interfere. Besides it, looks good enough.

bjornfor commented on pkgs/servers/authelia/update.sh:
```diff
@@ -0,0 +1,20 @@
+OLD_HTML_VERSION=$(nix-instantiate --eval --strict -E "with import ./. {}; authelia.version" | tr -d '"')
+OLD_HTML_SHA=$(fetch_html "${OLD_HTML_VERSION}")
+
+nix-update authelia
```
`nix-update` needs to be added to the nix-shell shebang line, right?

RaitoBezarius commented:
@06kellyjac I'm interested into using this and doing a NixOS module, can I do something to help you moving this PR forward?

06kellyjac commented:
Ill just rebase this on master today & try get the package merged first. If you make any progress on the module you can post here or message me on matrix.
All help on this is welcomed.

I can add you as a maintainer too if youd like. :)

RaitoBezarius commented:
Please do add me as a maintainer :)

Le mar. 30 août 2022 à 12:24, j-k ***@***.***> a écrit :

> Ill just rebase this on master today & try get the package merged first.
> If you make any progress on the module you can post here or message me on
> matrix.
> All help on this is welcomed.
>
> I can add you as a maintainer too if youd like. :)
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/NixOS/nixpkgs/pull/180469#issuecomment-1231470997>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AACMZRBBQC3IH7NTJS5BZCDV3XOONANCNFSM524UUMPQ>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
>

06kellyjac commented:
Result of `nixpkgs-review pr 180469` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>authelia</li>
  </ul>
</details>

RaitoBezarius commented on pkgs/servers/authelia/default.nix:
```diff
@@ -0,0 +1,81 @@
+
+let
+  version = "4.36.5";
+  webUiStatic = fetchurl {
```
I would move `webUiStatic` in the derivation so that end users can override it using `finalAttrs` and maybe put it as a `passthru` derivation too.

06kellyjac replied:
good point

06kellyjac replied:
can you give an example of how `finalAttrs` works?

LMK if the changes will work for that

RaitoBezarius replied:
```
mkDerivation (finalAttrs: {
 ui = stuff;
 passthru = { inherit (finalAttrs) ui; }
}
```

enable users to do `pkgs.authelia.overrideAttrs (old: { ui = newUi; }).ui == newUi`

RaitoBezarius replied:
It also remove the need for `rec`

06kellyjac replied:
but the `postPatch` would still need changing no? I'd have thought it'd use the old `ui` value

I guess I can cahnge the postPatch bit to use the bash var `$ui` rather than using the value directly from nix eval `${ui}` ?

RaitoBezarius replied:
function-style `mkDerivation` is really `rec` + "magic override support" (as long as you use `finalAttrs`)

RaitoBezarius reviewed (approved):
Sounds good to me.

RaitoBezarius commented:
@06kellyjac is it ready for review?

RaitoBezarius commented on pkgs/servers/authelia/default.nix:
```diff
@@ -0,0 +1,89 @@
+  '';
+
+  passthru = {
+    # if overriding replace the postFetch to put your UI config in internal/server/public_html
```
the `postPatch` you meant?

RaitoBezarius commented on pkgs/servers/authelia/default.nix:
```diff
@@ -0,0 +1,89 @@
+{ lib, fetchFromGitHub, buildGoModule, installShellFiles, fetchurl }:
+
+buildGoModule rec {
```
if you do use `finalAttrs` here, `ui` can be overriden safely without issue for `postPatch` and the rest, I think ; up to you :).

AndersonTorres replied:
Maybe it has no support to the new format yet.

RaitoBezarius replied:
It has no support for the new format, indeed!

RaitoBezarius reviewed (approved):
LGTM

06kellyjac commented:
We could merge it for now and make the changes later or wait.
Im away from my dev machine for a week or so, right now

Frostman commented:
LGTM, tried on my env with existing config, and seems to be working fine.

mweinelt commented:
Dump for a basic module that we've been building today. Needs work!

```
{ lib
, pkgs
, config
, ...
}:

let
  cfg = config.services.authelia;

  pkg = pkgs.callPackage ./package.nix { };

  format = pkgs.formats.yaml { };
  configFile = format.generate "config.yml" cfg.settings;
in
{
  options.services.authelia = with lib; {
    enable = mkEnableOption "Authelia";

    jwtSecretFile = mkOption {
      type = types.path;
      default = null;
      description = ''
        Path to your JWT secret used during identity verificaiton.
      '';
    };

    oidcIssuerPrivateKeyFile = mkOption {
      type = types.nullOr types.path;
      default = null;
      description = ''
        Path to your private key file used to encrypt OIDC JWTs.
      '';
    };

    oidcHmacSecretFile = mkOption {
      type = types.nullOr types.path;
      default = null;
      description = ''
        Path to your HMAC secret used to sign OIDC JWTs.
      '';
    };

    sessionSecretFile = mkOption {
      type = types.nullOr types.path;
      default = null;
      description = ''
        Path to your session secret. Only used when redis is used as session storage.
      '';
    };

    storageEncryptionKeyFile = mkOption {
      type = types.path;
      default = null;
      description = ''
        Path to your storage encryption key.
      '';
    };

    settings = mkOption {
      description = ''
        Your Authelia config.yml as a Nix attribute set.

        https://github.com/authelia/authelia/blob/master/config.template.yml
      '';
      type = types.submodule {
        freeformType = format.type;
        options = {
          default_2fa_method = mkOption {
            type = types.enum [ "" "totp" "webauthn" "mobile_push" ];
            default = "";
            example = "webauthn";
            description = ''
              Default 2FA method for new users and fallback for preferred but disabled methods.
            '';
          };

          server = {
            host = mkOption {
              type = types.str;
              default = "localhost";
              example = "0.0.0.0";
              description = ''
              '';
            };

            port = mkOption {
              type = types.port;
              default = 9091;
              description = ''
              '';
            };
          };

          log = {
            level = mkOption {
              type = types.enum [ "info" "debug" "trace" ];
              default = "info";
              example = "debug";
              description = ''
              '';
            };
          };
        };
      };
    };
  };

  config = lib.mkIf cfg.enable {

    environment.systemPackages = [
      pkg
    ];

    users.users.authelia = {
      isSystemUser = true;
      group = "authelia";
    };
    users.groups.authelia = {};

    systemd.services.authelia = {
      description = "Authelia authentication and authorization server";
      wantedBy = [
        "multi-user.target"
      ];
      environment = {
        AUTHELIA_JWT_SECRET_FILE = cfg.jwtSecretFile;
        AUTHELIA_STORAGE_ENCRYPTION_KEY_FILE = cfg.storageEncryptionKeyFile;
      } // lib.optionalAttrs (cfg.sessionSecretFile != null) {
        AUTHELIA_SESSION_SECRET_FILE = cfg.sessionSecretFile;
      } // lib.optionalAttrs (cfg.oidcIssuerPrivateKeyFile != null) {
        AUTHELIA_IDENTITY_PROVIDERS_OIDC_ISSUER_PRIVATE_KEY_FILE = cfg.oidcIssuerPrivateKeyFile;
        AUTHELIA_IDENTITY_PROVIDERS_OIDC_HMAC_SECRET_FILE = cfg.oidcHmacSecretFile;
      };
      serviceConfig = {
        User = "authelia";
        Group = "authelia";
        ExecStart = "${pkg}/bin/authelia --config ${configFile}";
      };
    };
  };
}

mweinelt commented:
Somewhat disappointing it doesn't support multiple webauthn devices at the same time.

06kellyjac commented:
Nice work

Ill be back in a couple of days

mweinelt commented:
> Somewhat disappointing it doesn't support multiple webauthn devices at the same time.

Feature-wise I'm thinking about looking into authentik next.

mweinelt commented:
Needs to be updated to 4.37.3 before this should be merged.

06kellyjac committed:
authelia: init at 4.37.5

Co-authored-by: Mostly Void <7rat13@gmail.com>

06kellyjac committed:
nixos/authelia: init module

Co-authored-by: Martin Weinelt <hexa@darmstadt.ccc.de>

hacker1024 commented:
> Needs to be updated to 4.37.3 before this should be merged.

4.37.5 now.

On another note, as I haven't seen any mention of this yet - @LongerHV has a low-level module up on [GitLab](https://gitlab.com/LongerHV/nixos-configuration/-/blob/master/modules/nixos/authelia.nix), which can be extended to add higher-level options but is very useful as-is as well.

hacker1024 commented:
Here is my attempt at adapting the aforementioned module to support multiple instances, which is a requirement for multiple domains:

```nix
# Based on https://gitlab.com/LongerHV/nixos-configuration/-/blob/3caea34bbe670b772aad8a1f3890665df509accb/modules/nixos/authelia.nix.

{ config, pkgs, lib, ... }:

let cfg = config.services.authelia;
in {
  options.services.authelia = {
    instances = lib.mkOption {
      type = with lib.types;
        attrsOf (submodule ({ name, ... }: {
          options = {
            enable = lib.mkEnableOption "Authelia instance";
            name = lib.mkOption {
              type = with lib.types; str;
              default = name;
            };
            package = lib.mkOption {
              default = pkgs.authelia;
              type = with lib.types; package;
            };
            environment = lib.mkOption {
              type = with lib.types; attrs;
              default = { };
            };
            settings = lib.mkOption {
              type = with lib.types; attrs;
              default = { };
            };
            settingsFile = lib.mkOption {
              type = with lib.types; str;
              default = "";
            };
            user = lib.mkOption {
              default = "authelia-${name}";
              type = with lib.types; str;
            };
            group = lib.mkOption {
              default = "authelia-${name}";
              type = with lib.types; str;
            };
          };
        }));
    };
  };

  config = let
    mkInstanceServiceConfig = instance: {
      description = "Authelia SSO service";
      wantedBy = [ "multi-user.target" ];
      after = [ "network.target" ];
      path = [ instance.package ];
      script = let
        generatedConfigArgument = lib.optionalString (instance.settings != { })
          "--config ${
            (pkgs.formats.yaml { }).generate "configuration.yaml"
            instance.settings
          }";
        configFileArgument = lib.optionalString (instance.settingsFile != "")
          "--config ${instance.settingsFile}";
      in "authelia ${generatedConfigArgument} ${configFileArgument}";
      serviceConfig = {
        User = instance.user;
        Group = instance.group;
      };
      inherit (instance) environment;
    };
    mkInstanceUsersConfig = instance: {
      groups."authelia-${instance.name}" =
        lib.mkIf (instance.group == "authelia-${instance.name}") {
          name = "authelia-${instance.name}";
        };
      users."authelia-${instance.name}" =
        lib.mkIf (instance.user == "authelia-${instance.name}") {
          name = "authelia-${instance.name}";
          isSystemUser = true;
          group = instance.group;
        };
    };
  in {
    systemd.services = lib.mkMerge (map (instance:
      lib.mkIf instance.enable {
        "authelia-${instance.name}" = mkInstanceServiceConfig instance;
      }) (builtins.attrValues cfg.instances));
    users = lib.mkMerge
      (map (instance: lib.mkIf instance.enable (mkInstanceUsersConfig instance))
        (builtins.attrValues cfg.instances));
  };
}
```

06kellyjac committed:
nixos/authelia: init tests

06kellyjac commented:
Did some updates based on the comments. Will finish off the docs tomorrow

LongerHV commented on nixos/modules/services/web-servers/authelia.nix:
```diff
@@ -0,0 +1,181 @@
+        type = types.str;
+      };
+
+      jwtSecretFile = mkOption {
```
I think a generic `environment` option would be better than defining all possible secret files this way.

06kellyjac replied:
I think a general environment option is good but I think a secrets section is still useful to promote setting them properly.
I'll maybe put these under `secrets.X` and have a separate environment and then merge the results

mweinelt replied:
It's also nice to know when you missed setting some, because enabling one might require another.

06kellyjac replied:
added environmentVariables. moved the secrets under `secrets.X`. Have the sanity check for the 2 required secrets but have an opt-out under `secrets.manual = true;`

LongerHV commented on nixos/modules/services/web-servers/authelia.nix:
```diff
@@ -0,0 +1,181 @@
+        '';
+      };
+
+      settings = mkOption {
```
`settings` option can be supplemented by an additional `settingsFile` option. It is useful, if you want to add client secrets, but don't want the to be visible in code.

https://github.com/LongerHV/nixos-configuration/blob/75f48d7d6853d355b8ad46cc1cda09436b97ed03/modules/nixos/authelia.nix#L25

06kellyjac replied:
Well the client secrets won't be within nix but the secrets will still be in the config file which is still not really recommended. I did intend to add a settingsFile option though.

https://www.authelia.com/configuration/methods/secrets

I'll add that now and I think I'll add a sanity check

LongerHV replied:
Unfortunatelly client secrets must be defined in the configuration file, see this [issue](https://github.com/authelia/authelia/issues/3249).
I have most configuration in plain nix, but client specific stuff is encrypted with agenix and passed to authelia as a second config file.

06kellyjac replied:
interesting. yeah at least you can provide authelia multiple config files (added a `settingsFiles` array)
just finishing up some bits

jsimonetti commented on nixos/modules/services/web-servers/authelia.nix:
```diff
@@ -0,0 +1,181 @@
+        description = "Authelia authentication and authorization server";
+        wantedBy = [ "multi-user.target" ];
+        after = [ "network.target" ];
+        serviceConfig = {
```
Just as a request from an outsider... 

Since this is an authentication platform, can we add some hardening to the systemd unit?

06kellyjac replied:
Of course.
The upstream doesn't have any but I don't see why we can't add some

https://github.com/authelia/authelia/blob/master/authelia.service

I'll grab some items from other nixos modules with hardened services

06kellyjac commented:
Fleshed out the docs, added some more items in the settings. Addressed most of the reviews.

Still need to add some hardening options to the service & add some notes to the nixos release docs

06kellyjac commented:
I think this is in a good position for final review. Anyone who wants to go into the maintainer list for the module too LMK

RaitoBezarius commented on nixos/modules/services/web-servers/authelia.nix:
```diff
@@ -0,0 +1,397 @@
+            StateDirectory = "authelia-${instance.name}";
+
+            # Security options:
+            AmbientCapabilities = "";
```
I don't see `ProtectSystem=strict|full`, did you try it with no avail?

06kellyjac replied:
Nope not tried it.

If you have any other values I should try LMK. Im not super au fait with systemd service hardening

06kellyjac replied:
Tried strict and it was happy at least in the vm test. Also added a couple more options from other areas. seemed to all run fine.
After some real world use we might need to tweak these :shrug:

RaitoBezarius commented on nixos/tests/authelia.nix:
```diff
@@ -0,0 +1,172 @@
+        };
+      };
+
+      documentation.nixos.enable = lib.mkForce false;
```
I wonder why this is not part of the default profile for NixOS tests?!

06kellyjac replied:
Ah this was for when I didnt have documentation and the vm tests were failing to build the docs.
This was caused by myself enabling the docs so I could serve them in my simple http service but I replaced that with a hello file instead.

Ill remove this

RaitoBezarius commented on nixos/tests/authelia.nix:
```diff
@@ -0,0 +1,172 @@
+
+    with subtest("Check for authelia"):
+      # expect the login page
+      assert "Login - Authelia" in authelia.succeed("curl --insecure -sSf -H Host:auth.example.com https://authelia:443/")
```
Please add error message to your asserts: `assert condition, "error in case assert is violated"`

06kellyjac replied:
Will do

RaitoBezarius commented on pkgs/servers/authelia/default.nix:
```diff
@@ -0,0 +1,110 @@
+  '';
+
+  passthru = {
+    # if overriding replace the postFetch to put your web UI config in internal/server/public_html
```
postPatch you meant?

06kellyjac replied:
Yeah, I moved it after migrating to a source build of the frontend. Will fix

RaitoBezarius reviewed (approved):
LGTM, modulo some nits/comments

06kellyjac commented on nixos/modules/services/web-servers/authelia.nix:
```diff
@@ -0,0 +1,397 @@
+            })
+            // instance.environmentVariables;
+
+          preStart = "${execCommand} ${configArg} validate-config";
```
Well we would have to implement every single setting and add assertions for conflicting/suddenly required values.

I have done some of that with the hard-required secrets that cause authelia to die if missing.

Also it wouldn't catch issues for cases using the catchall environmentVariables or configFiles options.
Nginx has a similar prestart config check

06kellyjac commented on nixos/modules/services/web-servers/authelia.nix:
```diff
@@ -0,0 +1,397 @@
+              services.authelia.${name}.secrets.storageEncryptionKeyFile = /my/path/to/encryptionkey;
+              Or set services.authelia.${name}.secrets.manual = true and provide them yourself via
+              environmentVariables or settingsFiles.
+              Do not include raw secrets in nix settings.
```
Do you mean remove line 380?
It's just an extra note just incase someone doesn't know, I dont think it causes any harm and might help a couple users

AndersonTorres commented on nixos/modules/services/web-servers/authelia.nix:
```diff
@@ -0,0 +1,397 @@
+
+  autheliaOpts = with lib; { name, ... }: {
+    options = {
+      enable = mkEnableOption (lib.mdDoc "Authelia instance");
```
> with lib
> lib.mdDoc

Another day, another inexplicable idiosyncrasy.

06kellyjac replied:
And I guess you've never missed one line 🙃

hacker1024 commented:
> 

This module looks great! I'd be happy to help maintain.

AndersonTorres commented on nixos/tests/authelia.nix:
```diff
@@ -0,0 +1,172 @@
+# Test Traefik as a reverse proxy of a local web service
+# and a Docker container.
+import ./make-test-python.nix ({ pkgs, ... }: {
+  name = "authelia";
+  meta = with pkgs.lib.maintainers; {
+    maintainers = [ jk ];
+  };
```
```suggestion
import ./make-test-python.nix ({ lib, pkgs, ... }: {
  name = "authelia";
  meta.maintainers = [ pkgs.lib.maintainers.jk ];
};
```

mweinelt replied:
Please use `meta.maintainers = with lib.maintainers; [ jk ];` after importing lib directly. Using `with` in such a limited scope is not bad per se, and allows more maintainers to easily add/remove themselves without carrying the scoping around.

AndersonTorres replied:
Is there anyone else interested in maintaining this module? O.O

Imho the maintainers of the package should automatically be maintainers of modules, but there are some practical problems.

AndersonTorres commented on pkgs/servers/authelia/default.nix:
```diff
@@ -0,0 +1,110 @@
+    rev = "v${version}";
+    sha256 = "sha256-xsdBnyPHFIimhp2rcudWqvVR36WN4vBXbxRmvgqMcDw=";
+  };
+  vendorSha256 = "sha256-mzGE/T/2TT4+7uc2axTqG3aeLMnt1r9Ya7Zj2jIkw/w=";
```
```suggestion

  vendorSha256 = "sha256-mzGE/T/2TT4+7uc2axTqG3aeLMnt1r9Ya7Zj2jIkw/w=";
```

AndersonTorres commented on pkgs/servers/authelia/default.nix:
```diff
@@ -0,0 +1,110 @@
+      runHook postInstall
+    '';
+  };
+
```
```suggestion
```

AndersonTorres commented on pkgs/servers/authelia/default.nix:
```diff
@@ -0,0 +1,110 @@
+  nativeBuildInputs = [ installShellFiles ];
+
+  web = buildNpmPackage {
+    inherit src version;
+
+    pname = "authelia-web";
+    sourceRoot = "source/web";
```
```suggestion
    pname = "authelia-web";
    inherit version src;

    prePatch = ''
      cd source/web
    '';
```

06kellyjac replied:
what is the benefit here?

AndersonTorres replied:
Setting `sourceRoot` is too low-level, it isn't suitable to be exposed as if it was an imperative variable.

RaitoBezarius replied:
What? I think we do this kind of things in a lot of place, do you have an authoritative source (e.g. recent commits / PRs where we cleaned up this pattern) on that?

AndersonTorres replied:
No.

AndersonTorres commented on pkgs/servers/authelia/default.nix:
```diff
@@ -0,0 +1,110 @@
+
+  nativeBuildInputs = [ installShellFiles ];
+
+  web = buildNpmPackage {
```
P.S.: I believe you should put this on a separate file.

06kellyjac committed:
authelia: split out main and web to separate files

mweinelt commented on nixos/modules/module-list.nix:
```diff
@@ -1212,6 +1212,7 @@
   ./services/web-apps/zabbix.nix
   ./services/web-servers/agate.nix
   ./services/web-servers/apache-httpd/default.nix
+  ./services/web-servers/authelia.nix
```
Authelia is certainly not a webserver.

06kellyjac replied:
It responds to http requests, it also has a web ui.

I can move it under security if you want, but `web-apps` (like keycloak) sounds bad

mweinelt replied:
Yeah, security is probably a better spot. For the package I would think of `pkgs/servers/security/authelia` as a good place.

RaitoBezarius commented:
@AndersonTorres I think there were still unclosed comments, for example, @mweinelt's one. Please do not do this again.

AndersonTorres commented:
Wasn't this already approved by three people including you, @RaitoBezarius?

figsoda commented:
The issue is that there is unresolved (and fresh) conversion that needs to be addressed

AndersonTorres commented:
Approvals are not so useful as I thought, so.
Sorry for the bad judgement.

RaitoBezarius commented:
@06kellyjac could we bother you with sending a follow up PR to move it under security? I do agree with @mweinelt comment.

dotlambda commented:
Can we downgrade this to 4.31.0? Back then authelia had a `yarn.lock` and we could use `fetchYarnDeps` rather than vendor 16k lines.

LongerHV commented:
@dotlambda Downgrade would braek people's systems (including mine), as database migration downgrades are not really possible (and certainly are not automatic). This version of authelia is already present in NixOS 23.05...

dotlambda commented:
Then please find another alternative to vendoring 16k lines.
